### DISCUSSION TITLE
Allow Bundles to be added to storefront sales channels

### DISCUSSION THREAD

**Discussion by pauldemarco** [Shopify/storefront-api-feedback#206] [Help]
Using the Shopify Bundle App, it seems I am unable to publish the new bundle to anything other than the “Online Store”. 

I’d like this bundle product to be available via the Storefront API so I can use it in my headless app.

**Comment by marnauortega** [comment]
I suppose that at this point, you've been able to add it but otherwise, did you try enabling it on the headless channel? Did you make sure that it was not in draft mode, which is how it starts by default?

**Comment by vamshiaruru** [comment]
I am facing the same issue. On one store I am able to publish on an app sale channel, aon another store I am not able to. This app has all accesses, so I am not sure what is happening.

**Comment by siddhantbajaj** [comment]
3p channels apps needs to enable bundles feature from their partner dashboard. Here's how how partners can do it - https://shopify.dev/docs/apps/selling-strategies/bundles/enable-publishing. I'll close this but feel free to reopen or ask any related question.

**Reply by pedropedruzzi** [reply]
That's cool, thanks. However this can only be enabled for sales channel apps (which must be public apps), except for admin-created custom apps, for which publishing bundles is enabled by default. Apparently custom apps form Partners were left out of the bundles' party.

**Comment by ishaanchawla** [comment]
Hi there. Bundles can be sold through the Storefront API now (enabling Hydrogen storefronts and Headless apps), and support for this was shipped last year. Furthermore, if there is a 3P sales channel app that wants to be able to sell a Bundle, they can mark themselves as eligible to sell bundles as shown [here](https://shopify.dev/docs/apps/selling-strategies/bundles/enable-publishing).
